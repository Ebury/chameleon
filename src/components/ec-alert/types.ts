export enum PropsAlertType {
  TEXT = 'text',
  DATE = 'date',
  NUMBER = 'number',
  TEL = 'tel'
}

export enum AlertType {
    ERROR = 'error', 
    INFO = 'info', 
    SUCCESS = 'success', 
    WARNING = 'warning'
}

export enum AlertEvent {
  UPDATE_OPEN = 'update:open',
  ACTION = 'action',
  CHANGE = 'change',
}

export interface AlertEvents {
  [AlertEvent.UPDATE_OPEN]: AlertProps['open']
  [AlertEvent.ACTION]: undefined
  [AlertEvent.CHANGE]: AlertProps['open']
}

export interface AlertProps {
  propsType?: PropsAlertType,
  type: AlertType,
  title: string,
  subtitle?: string,
  dismissable?: boolean,
  buttonText?: string,
  open?: boolean,
  responsive?: boolean
}
