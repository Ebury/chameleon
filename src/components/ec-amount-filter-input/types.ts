import type { DropdownItem } from '../ec-dropdown/types';

export interface ComparisonSymbolItem extends DropdownItem<string> {}

export interface AmountFilterInputModel {
  amount?: number,
  comparisonSymbol?: ComparisonSymbolItem,
}

export interface AmountFilterInputProps {
  modelValue: AmountFilterInputModel,
  locale?: string,
  isSensitive?: boolean,
  label?: string,
  note?: string,
  amountPlaceholder?: string,
  bottomNote?: string,
  isWarning?: boolean,
  warningTooltipMessage?: string,
  errorMessage?: string,
  errorTooltipMessage?: string,
  comparisonSymbolItems: ComparisonSymbolItem[],
}
