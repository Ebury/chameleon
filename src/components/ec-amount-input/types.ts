import type { InputFieldProps } from '../ec-input-field/types';

export interface AmountInputProps extends Omit<InputFieldProps, 'modelValue'> {
  modelValue?: number | string | null,
  isMasked?: boolean,
  locale?: string,
  currency?: string,
}
