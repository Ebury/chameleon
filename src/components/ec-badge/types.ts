import { AlertType as BadgeType } from '../ec-alert/types';

export { BadgeType };

export interface BadgeProps {
  type?: BadgeType,
  value: string | number,
}
