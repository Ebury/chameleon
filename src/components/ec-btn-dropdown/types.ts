import type { AnchorHTMLAttributes } from 'vue';
import type { RouteLocationRaw } from 'vue-router';

import type { DropdownItem } from '../ec-dropdown-search/types';

export interface BtnDropdownItem<TValue = string> extends DropdownItem<TValue> {
  href?: string,
  to?: string | RouteLocationRaw,
  attrs?: AnchorHTMLAttributes,
}

export interface BtnDropdownProps<TValue = string, TBtnDropdownItem extends BtnDropdownItem<TValue> = BtnDropdownItem<TValue>> {
  items?: TBtnDropdownItem[],
  isBtnDropdownDisabled?: boolean,
  buttonText: string,
  listDataTest?: string,
  href?: string,
  to?: string | RouteLocationRaw,
}
