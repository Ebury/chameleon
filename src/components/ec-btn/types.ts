import type { RouteLocationRaw } from 'vue-router';

import type { IconName } from '../ec-icon/iconNames';

export enum ButtonSize {
  Medium = 'md',
  Small = 'sm'
}

export enum ButtonCategory {
  Primary = 'primary',
  Secondary = 'secondary',
  Success = 'success',
  Error = 'error',
  Warning = 'warning'
}

export interface ButtonProps {
  size?: ButtonSize;
  href?: string;
  to?: RouteLocationRaw;
  tag?: string;
  isDisabled?: boolean;
  icon?: IconName;
  isRounded?: boolean;
  isOutline?: boolean;
  isFullWidth?: boolean;
  category: ButtonCategory;
  isReverse?: boolean;
  isLoading?: boolean;
  isSubmit?: boolean
}
