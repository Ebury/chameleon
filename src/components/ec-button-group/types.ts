export interface ButtonGroupItem<TValue> {
  id?: string | number,
  value: TValue,
  text: string,
  disabled?: boolean,
}

export interface ButtonGroupProps<TValue> {
  modelValue?: TValue,
  items: ButtonGroupItem<TValue>[],
}
