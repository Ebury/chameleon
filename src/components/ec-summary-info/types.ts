import type { IconName } from '../ec-icon/types';

export enum StylePreset {
    LABEL = 'label',
    DESCRIPTION = 'description',
    TEXT = 'text'
}
export interface Item {
  text?: string,
  tooltipText?: string,
  stylePreset?: StylePreset,
  isSensitive?: boolean
}

export interface SummaryProps {
    iconName?: IconName,
    lineItems: Item[]
  }
