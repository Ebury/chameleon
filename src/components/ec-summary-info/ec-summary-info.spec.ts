import type { MountingOptions } from '@vue/test-utils';
// eslint-disable-next-line import/no-extraneous-dependencies
import { mount } from '@vue/test-utils';

import type { CVueWrapper } from '../../../tests/utils/global';
import { IconName } from '../ec-icon/types';
import EcSummaryInfo from './ec-summary-info.vue';
import type { SummaryProps } from './types';
import { StylePreset } from './types';

describe('EcSummaryInfo', () => {
  function mountSummaryInfo(props?: Partial<SummaryProps>, mountOpts?: MountingOptions<SummaryProps>) {
    return mount(EcSummaryInfo as any, // eslint-disable-line 
      {
        props: {
          lineItems: [],
          ...props,
        },
        ...mountOpts,
      },
    ) as CVueWrapper;
  }

  describe('icon', () => {
    it('should render with the given "icon" prop', () => {
      const wrapper = mountSummaryInfo({ iconName: IconName.SimpleSell });
      expect(wrapper.findByDataTest('ec-summary-info__main-icon').exists()).toBe(true);
      expect(wrapper.element).toMatchSnapshot();
    });

    it('should not render with the not given "icon" prop', () => {
      const wrapper = mountSummaryInfo();
      expect(wrapper.findByDataTest('ec-summary-info__main-icon').exists()).toBe(false);
    });
  });

  describe('lineItems', () => {
    it('should render with the given "lineItems" prop', () => {
      const wrapper = mountSummaryInfo({
        lineItems: [
          {
            stylePreset: StylePreset.LABEL,
            text: 'The label',
          },
          {
            stylePreset: StylePreset.TEXT,
            text: 'Some text',
            tooltipText: 'Some tooltip text',
          },
          {
            stylePreset: StylePreset.DESCRIPTION,
            text: 'Another text',
          },
        ],
      });
      expect(wrapper.findByDataTest('ec-summary-info__content').exists()).toBe(true);
      expect(wrapper.element).toMatchSnapshot();
    });

    it('should render with the not given "lineItems" prop', () => {
      const wrapper = mountSummaryInfo();
      expect(wrapper.findByDataTest('ec-summary-info__content-line-item-content-label').exists()).toBe(false);
      expect(wrapper.findByDataTest('ec-summary-info__content-line-item-content-text').exists()).toBe(false);
      expect(wrapper.findByDataTest('ec-summary-info__content-line-item-icon-text').exists()).toBe(false);
      expect(wrapper.findByDataTest('ec-summary-info__content-line-item-content-description').exists()).toBe(false);
      expect(wrapper.findByDataTest('ec-summary-info__content-line-item-icon-description').exists()).toBe(false);
      expect(wrapper.element).toMatchSnapshot();
    });

    it('should render with default stylePresets', () => {
      const wrapper = mountSummaryInfo({
        lineItems: [
          {
            text: StylePreset.LABEL,
          },
          {
            text: StylePreset.TEXT,
            tooltipText: 'Some tooltip text',
          },
          {
            text: StylePreset.DESCRIPTION,
          },
          {
            text: 'some extra text',
          },
        ],
      });

      expect(wrapper.findByDataTest('ec-summary-info__content').element.children[0].children[0]).toMatchSnapshot();
      expect(wrapper.findByDataTest('ec-summary-info__content').element.children[1].children[0]).toMatchSnapshot();
      expect(wrapper.findByDataTest('ec-summary-info__content').element.children[2].children[0]).toMatchSnapshot();
      expect(wrapper.findByDataTest('ec-summary-info__content').element.children[3].children[0]).toMatchSnapshot();
    });

    describe('label', () => {
      describe('when the label is present', () => {
        it('should render the label', () => {
          const wrapper = mountSummaryInfo({
            lineItems: [
              {
                stylePreset: StylePreset.LABEL,
                text: 'testing the label',
              },
            ],
          });

          expect(wrapper.findByDataTest('ec-summary-info__content-line-item-content-label').exists()).toBe(true);
          expect(wrapper.findByDataTest('ec-summary-info__content-line-item-content-label').text()).toBe('testing the label');
        });
      });
    });

    describe('text', () => {
      describe('when the text is present', () => {
        it('should render the text', () => {
          const wrapper = mountSummaryInfo({
            lineItems: [
              {
                stylePreset: StylePreset.TEXT,
                text: 'testing the text',
              },
            ],
          });
          expect(wrapper.findByDataTest('ec-summary-info__content-line-item-content-text').exists()).toBe(true);
          expect(wrapper.findByDataTest('ec-summary-info__content-line-item-content-text').text()).toBe('testing the text');
        });
      });

      describe('when the text has a tooltip', () => {
        it('should render with a tooltip', () => {
          const wrapper = mountSummaryInfo({
            lineItems: [
              {
                stylePreset: StylePreset.TEXT,
                tooltipText: 'Some tooltip text',
              },
            ],
          });
          expect(wrapper.findByDataTest('ec-tooltip-mock').exists()).toBe(true);
          expect(wrapper.findByDataTest('ec-tooltip-mock')).toMatchSnapshot();
        });

        it('should render an info icon', () => {
          const wrapper = mountSummaryInfo({
            lineItems: [
              {
                stylePreset: StylePreset.TEXT,
                tooltipText: 'Some tooltip text',
              },
            ],
          });
          expect(wrapper.findByDataTest('ec-summary-info__content-line-item-icon-text').exists()).toBe(true);
        });
      });
    });

    describe('description', () => {
      describe('when the description is present', () => {
        it('should render the description', () => {
          const wrapper = mountSummaryInfo({
            lineItems: [
              {
                stylePreset: StylePreset.DESCRIPTION,
                text: 'testing the description',
              },
            ],
          });
          expect(wrapper.findByDataTest('ec-summary-info__content-line-item-content-description').exists()).toBe(true);
          expect(wrapper.findByDataTest('ec-summary-info__content-line-item-content-description').text()).toBe('testing the description');
        });
      });

      describe('when the description has a tooltip', () => {
        it('should render with a tooltip', () => {
          const wrapper = mountSummaryInfo({
            lineItems: [
              {
                stylePreset: StylePreset.DESCRIPTION,
                tooltipText: 'Some tooltip text',
              },
            ],
          });
          expect(wrapper.findByDataTest('ec-tooltip-mock').exists()).toBe(true);
          expect(wrapper.findByDataTest('ec-tooltip-mock')).toMatchSnapshot();
        });

        it('should render an info icon', () => {
          const wrapper = mountSummaryInfo({
            lineItems: [
              {
                stylePreset: StylePreset.DESCRIPTION,
                tooltipText: 'Some tooltip text',
              },
            ],
          });
          expect(wrapper.findByDataTest('ec-summary-info__content-line-item-icon-description').exists()).toBe(true);
        });
      });

      describe('when the description isSensitive', () => {
        it('should render description and tooltip with a sensitive class', () => {
          const wrapper = mountSummaryInfo({
            lineItems: [
              {
                stylePreset: StylePreset.DESCRIPTION,
                text: 'Some text',
                isSensitive: true,
              },
            ],
          });
          expect(wrapper.findByDataTest('ec-summary-info__content-line-item-content-description').classes('my-sensitive-content-test-class')).toBe(true);
          expect(wrapper.element).toMatchSnapshot();
        });
      });
    });
  });

  describe('slots', () => {
    it('should render with the default slot given', () => {
      const wrapper = mountSummaryInfo({ iconName: IconName.SimpleSell }, {
        slots: {
          default: `<div data-test="default-slot">
            <span>
              Some text from a slot
            </span>
          </div>`,
        },
      });
      expect(wrapper.findByDataTest('default-slot').exists()).toBe(true);
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
