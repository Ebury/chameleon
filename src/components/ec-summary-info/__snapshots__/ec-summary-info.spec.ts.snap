// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EcSummaryInfo icon should render with the given "icon" prop 1`] = `
<div
  class="ec-summary-info"
  data-test="ec-summary-info"
>
  <svg
    class="ec-icon ec-summary-info__main-icon"
    data-test="ec-summary-info__main-icon"
    height="16"
    width="16"
  >
    <use
      xlink:href="#ec-simple-sell"
    />
  </svg>
  
  <div
    class="ec-summary-info__content"
    data-test="ec-summary-info__content"
  >
    
    
  </div>
  
</div>
`;

exports[`EcSummaryInfo lineItems description when the description has a tooltip should render with a tooltip 1`] = `
DOMWrapper {
  "isDisabled": [Function],
  "wrapperElement": <svg
    class="ec-icon ec-summary-info__content-line-item-icon ec-summary-info__content-line-item-icon-description"
    data-ec-tooltip-mock-content="Some tooltip text"
    data-test="ec-summary-info__content-line-item-icon ec-summary-info__content-line-item-icon-description ec-mock ec-tooltip-mock"
    height="14"
    width="14"
  >
    <use
      xlink:href="#ec-simple-info"
    />
  </svg>,
}
`;

exports[`EcSummaryInfo lineItems description when the description isSensitive should render description and tooltip with a sensitive class 1`] = `
<div
  class="ec-summary-info"
  data-test="ec-summary-info"
>
  <!--v-if-->
  
  <div
    class="ec-summary-info__content"
    data-test="ec-summary-info__content"
  >
    
    <div
      class="ec-summary-info__content-line-item"
      data-test="ec-summary-info__content-line-item-0"
    >
      <span
        class="ec-summary-info__content-line-item-content ec-summary-info__content-line-item-content-description my-sensitive-content-test-class"
        data-test="ec-summary-info__content-line-item-content ec-summary-info__content-line-item-content-description"
        title="Some text"
      >
        Some text
      </span>
      <!--v-if-->
    </div>
    
  </div>
  
</div>
`;

exports[`EcSummaryInfo lineItems should render with default stylePresets 1`] = `
<span
  class="ec-summary-info__content-line-item-content ec-summary-info__content-line-item-content-label"
  data-test="ec-summary-info__content-line-item-content ec-summary-info__content-line-item-content-label"
  title="label"
>
  label
</span>
`;

exports[`EcSummaryInfo lineItems should render with default stylePresets 2`] = `
<span
  class="ec-summary-info__content-line-item-content ec-summary-info__content-line-item-content-text"
  data-test="ec-summary-info__content-line-item-content ec-summary-info__content-line-item-content-text"
  title="text"
>
  text
</span>
`;

exports[`EcSummaryInfo lineItems should render with default stylePresets 3`] = `
<span
  class="ec-summary-info__content-line-item-content ec-summary-info__content-line-item-content-description"
  data-test="ec-summary-info__content-line-item-content ec-summary-info__content-line-item-content-description"
  title="description"
>
  description
</span>
`;

exports[`EcSummaryInfo lineItems should render with default stylePresets 4`] = `
<span
  class="ec-summary-info__content-line-item-content ec-summary-info__content-line-item-content-text"
  data-test="ec-summary-info__content-line-item-content ec-summary-info__content-line-item-content-text"
  title="some extra text"
>
  some extra text
</span>
`;

exports[`EcSummaryInfo lineItems should render with the given "lineItems" prop 1`] = `
<div
  class="ec-summary-info"
  data-test="ec-summary-info"
>
  <!--v-if-->
  
  <div
    class="ec-summary-info__content"
    data-test="ec-summary-info__content"
  >
    
    <div
      class="ec-summary-info__content-line-item"
      data-test="ec-summary-info__content-line-item-0"
    >
      <span
        class="ec-summary-info__content-line-item-content ec-summary-info__content-line-item-content-label"
        data-test="ec-summary-info__content-line-item-content ec-summary-info__content-line-item-content-label"
        title="The label"
      >
        The label
      </span>
      <!--v-if-->
    </div>
    <div
      class="ec-summary-info__content-line-item"
      data-test="ec-summary-info__content-line-item-1"
    >
      <span
        class="ec-summary-info__content-line-item-content ec-summary-info__content-line-item-content-text"
        data-test="ec-summary-info__content-line-item-content ec-summary-info__content-line-item-content-text"
        title="Some text"
      >
        Some text
      </span>
      <svg
        class="ec-icon ec-summary-info__content-line-item-icon ec-summary-info__content-line-item-icon-text"
        data-ec-tooltip-mock-content="Some tooltip text"
        data-test="ec-summary-info__content-line-item-icon ec-summary-info__content-line-item-icon-text ec-mock ec-tooltip-mock"
        height="14"
        width="14"
      >
        <use
          xlink:href="#ec-simple-info"
        />
      </svg>
    </div>
    <div
      class="ec-summary-info__content-line-item"
      data-test="ec-summary-info__content-line-item-2"
    >
      <span
        class="ec-summary-info__content-line-item-content ec-summary-info__content-line-item-content-description"
        data-test="ec-summary-info__content-line-item-content ec-summary-info__content-line-item-content-description"
        title="Another text"
      >
        Another text
      </span>
      <!--v-if-->
    </div>
    
  </div>
  
</div>
`;

exports[`EcSummaryInfo lineItems should render with the not given "lineItems" prop 1`] = `
<div
  class="ec-summary-info"
  data-test="ec-summary-info"
>
  <!--v-if-->
  
  <div
    class="ec-summary-info__content"
    data-test="ec-summary-info__content"
  >
    
    
  </div>
  
</div>
`;

exports[`EcSummaryInfo lineItems text when the text has a tooltip should render with a tooltip 1`] = `
DOMWrapper {
  "isDisabled": [Function],
  "wrapperElement": <svg
    class="ec-icon ec-summary-info__content-line-item-icon ec-summary-info__content-line-item-icon-text"
    data-ec-tooltip-mock-content="Some tooltip text"
    data-test="ec-summary-info__content-line-item-icon ec-summary-info__content-line-item-icon-text ec-mock ec-tooltip-mock"
    height="14"
    width="14"
  >
    <use
      xlink:href="#ec-simple-info"
    />
  </svg>,
}
`;

exports[`EcSummaryInfo slots should render with the default slot given 1`] = `
<div
  class="ec-summary-info"
  data-test="ec-summary-info"
>
  <svg
    class="ec-icon ec-summary-info__main-icon"
    data-test="ec-summary-info__main-icon"
    height="16"
    width="16"
  >
    <use
      xlink:href="#ec-simple-sell"
    />
  </svg>
  
  <div
    data-test="default-slot"
  >
    <span>
       Some text from a slot 
    </span>
  </div>
  
</div>
`;
